### PR TITLE
use nested `Diffable`s for equivalence in stdlib data classes

### PR DESCRIPTION
**Pull request by tek**
I left the stacktrace stuff untouched, not sure this would be sensible.

### CODE REVIEW

File: matcher/src/main/scala/org/specs2/matcher/describe/Diffables.scala [hunk 1]
```diff
@@ -18,95 +18,104 @@ object PrimitiveDiffable {
 
 }
 
-class EitherDiffable[L : Diffable, R : Diffable](implicit ldi: Diffable[L],
-                                                 rdi: Diffable[R])
+class EitherDiffable[L: Diffable, R :Diffable]
```
Comment (etorreborre) [inline matcher/src/main/scala/org/specs2/matcher/describe/Diffables.scala]: Could you please maintain the formatting as `L : Diffable, R : Diffable`?

Comment (etorreborre) [inline matcher/src/main/scala/org/specs2/matcher/describe/Diffables.scala]: And also I should be more careful reading PRs, I didn't notice the redundancy in implicit params there.

**Comment by etorreborre** [comment]
That looks good, thanks.

**Comment by tek** [comment]
it actually produced a compile error because I use the implicit as a parameter again :)

**Pull request merged**